scipy/special/special_c_wrappers.cpp

```diff
@@ -0,0 +1,204 @@
+/* These wrappers exist to allow for calling functions from the under development
+ * header only C++ special function library in scipy/special/special within files
+ * for special functions which are still implemented in C or Cython. The hope is
+ * that these  wrappers will become unnecessary if all of special is translated
+ * into C++ */
+
+extern "C" {
+
+#include <numpy/npy_math.h>
+#include "special_c_wrappers.h"
```

Comment:
Same comment about `include` and `extern C`. I think it's bad practice to wrap an `include` in `extern "C"`. If it's "numpy/npy_math.h", we probably have no choice because it's a C file. If it's our own, like "special_c_wrappers.h", we should just make it multilingual.